icgcget/clients/gnos/gnos_client.py

```diff
@@ -17,34 +19,53 @@
 #
 
 import re
-import tempfile
+import os
+import shutil
 from icgcget.clients.download_client import DownloadClient
 from icgcget.clients.errors import SubprocessError
 
 
 class GnosDownloadClient(DownloadClient):
 
-    def __init__(self, json_path=None):
-        super(GnosDownloadClient, self) .__init__(json_path)
+    def __init__(self, json_path=None, docker=False, log_dir=None):
+        super(GnosDownloadClient, self).__init__(json_path, docker, log_dir)
         self.repo = 'cghub'
 
-    def download(self, uuids, access, tool_path, output, processes, udt=None, file_from=None, repo=None, password=None):
-        access_file = tempfile.NamedTemporaryFile()
-        access_file.file.write(access)
-        access_file.file.seek(0)
-        call_args = [tool_path, '-vv', '-c', access_file.name, '-d']
-        call_args.extend(uuids)
-        call_args.extend(['-p', output])
+    def download(self, uuids, access, tool_path, staging, processes, udt=None, file_from=None, repo=None,
+                 password=None):
+        access_file = self.get_access_file(access, staging)
+        call_args = []
+        if self.docker:
+            access_path = self.docker_mnt + '/' + os.path.split(access_file.name)[1]
+            # Client needs to be run using sh to be able to download files in docker container.
+            call_args = ['/bin/sh', '-c', tool_path + '-vv' + '-d ' + ' '.join(uuids) + ' -c ' + access_path +
+                         ' -p ' + self.docker_mnt + ' -l ' + staging + '/gnos_logs']
+            call_args = self.prepend_docker_args(call_args, staging)
+        else:
+            call_args.extend([tool_path, '-vv', '-d'])
+            call_args.extend(uuids)
+            call_args.extend(['-c', access_file.name, '-p', staging, '-l', self.log_dir + '/gnos_logs'])
         code = self._run_command(call_args, self.download_parser)
+        if self.docker:
+            shutil.move(staging + '/gnos_log', self.log_dir + '/gnos_log')
         return code
 
     def access_check(self, access, uuids=None, path=None, repo=None, output=None, api_url=None, password=None):
-        access_file = tempfile.NamedTemporaryFile()
-        access_file.file.write(access)
-        call_args = [path, '-vv', '-c', access_file.name, '-d']
-        call_args.extend(uuids)
-        call_args.extend(['-p', output])
+
+        access_file = self.get_access_file(access, output)
+        call_args = []
+        if self.docker:
+            access_path = self.docker_mnt + '/' + os.path.split(access_file.name)[1]
```

Comment:
There looks like a lot of copy paste code here. What's different between these sections? Looks like good opportunity for reuse.